app/hooks/header.ts

```diff
@@ -52,13 +52,13 @@ export const useCollapsibleHeader = <T>(isLargeTitle: boolean, onSnap?: (offset:
     const animatedRef = useAnimatedRef<Animated.ScrollView>();
     const {largeHeight, defaultHeight, headerOffset} = useHeaderHeight();
     const scrollValue = useSharedValue(0);
-    const lockValue = useSharedValue<number | null>(null);
+    const [lockValue, setLockValue] = useState<number>(0);
```

Comment:
why is this using state instead of the shared value?

Reply:
I've actually responded [here](https://github.com/mattermost/mattermost-mobile/pull/8026#discussion_r1645601219).

But, this is the main reason why the "Search" title is resting behind the input text.

Mostly because misuse of shared value and shared value on JS thread lags behind UI thread.  So, if we re-assign shared value, JS will not see the effect immediately.

Reply:
And lockValue when was being used as a shared value, was not to trigger any animation.  Just to hold a direct assignment.